scripts/jenkins/cloud/manual/run-deployer.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+set -u -x
+
+CLI=${CLI:=docker}
+# CLI=podman
+
+basedir=$(readlink --canonicalize $(dirname $0)/../../../../)
+
+ionice -c idle ${CLI} build --tag ci-deployer \
```

Comment:
In my ideal (perhaps fantasy) world, the container image is portable and doesn't need to be built. Instead, the workflow here would be:

1. docker pull <image name>
2. docker run ....

Things like git user name, etc. can be passed as environment variables and an entrypoint script in the contain can handle things like setting git user name, email, venv activation, etc.

Reply:
whatever the workflow ends up being, probably best to add to docs. specially if any environmental variables need to be set.  Can such things be pull automatically from git globa config?

Reply:
One thing I did was to volume mount my ~/.ssh and my gitconfig into the container. So there are 2 ways to approach this:

A) Pass all the needed info into the container as environment variables and have the entrypoint script handle everything (git setup, cloning, venv activation, etc.)
B) Have the entrypoint script assume everything (ie ssh & git settings, git repo, etc.) is already in place via volume mounts

Reply:
Mounting the `.ssh` directory is a nice idea which had not occurred to me. I have a patch for the `setup-ssh-access.yml` playbook to generate a new key if the deployer doesn't have one already, which would not be necessary if the `.ssh` directory was mounted.

I don't quite see though how I would mount the git configuration. The git config is split into several locations, i.e. system, global, and local and we would have to mount all of them to make sure the correct configuration is used inside the container.

Instead the `run-deployer.sh` wrapper uses git itself to tell it what user and email are to configure git inside the container. This approach is limited to known config keys though which might make it not flexible enough in the end.

Do you have config options besides user.name and user.email you would like to have in the container?